profitbricks-rest/src/main/java/org/apache/jclouds/profitbricks/rest/compute/ProfitBricksComputeServiceAdapter.java

```diff
@@ -0,0 +1,570 @@
+           @Named(TIMEOUT_NODE_RUNNING) Predicate<VolumeRef> waitVolumeUntilAvailable,
+           @Named(PROPERTY_USER_THREADS) ListeningExecutorService executorService,
+           @Named(POLL_PREDICATE_SERVER) Predicate<ServerRef> waitServerUntilAvailable,
+           @Named(TIMEOUT_NODE_RUNNING) Predicate<ServerRef> waitServerUntilRunning,
+           @Named(TIMEOUT_NODE_SUSPENDED) Predicate<ServerRef> waitServerUntilSuspended,
+           @Named(POLL_PREDICATE_NIC) Predicate<NicRef> waitNICUntilAvailable,
+           ProvisioningJob.Factory jobFactory,
+           ProvisioningManager provisioningManager) {
+      this.api = api;
+      this.waitDcUntilAvailable = waitDcUntilAvailable;
+      this.waitVolumeUntilAvailable = waitVolumeUntilAvailable;
+      this.waitServerUntilAvailable = waitServerUntilAvailable;
+      this.waitNICUntilAvailable = waitNICUntilAvailable;
+      this.waitServerUntilSuspended = waitServerUntilSuspended;
+      this.waitServerUntilRunning = waitServerUntilRunning;
+      this.executorService = executorService;
+      this.jobFactory = jobFactory;
+      this.provisioningManager = provisioningManager;
+      this.datacetners = ImmutableList.of();
+   }
+
+   @Override
+   public NodeAndInitialCredentials<ServerInDataCenter> createNodeWithGroupEncodedIntoName(String group, String name, Template template) {
+      checkArgument(template instanceof TemplateWithDataCenter, "This implementation requires a TemplateWithDataCenter");
+      return createNodeWithGroupEncodedIntoName(group, name, TemplateWithDataCenter.class.cast(template));
+   }
+
+   protected NodeAndInitialCredentials<ServerInDataCenter> createNodeWithGroupEncodedIntoName(String group, String name, TemplateWithDataCenter template) {
+      checkArgument(template.getLocation().getScope() == LocationScope.ZONE, "Template must use a ZONE-scoped location");
+      final String dataCenterId = template.getDataCenter().id();
+      Hardware hardware = template.getHardware();
+      TemplateOptions options = template.getOptions();
+      final String loginUser = isNullOrEmpty(options.getLoginUser()) ? "root" : options.getLoginUser();
+      final String password = options.hasLoginPassword() ? options.getLoginPassword() : Passwords.generate();
+      final org.jclouds.compute.domain.Image image = template.getImage();
+
+      // provision all volumes based on hardware
+      List<? extends Volume> volumes = hardware.getVolumes();
+      List<String> volumeIds = Lists.newArrayListWithExpectedSize(volumes.size());
+
+      int i = 1;
+      for (final Volume volume : volumes) {
+         try {
+            logger.trace("<< provisioning volume '%s'", volume);
+            final org.apache.jclouds.profitbricks.rest.domain.Volume.Request.CreatePayload request = org.apache.jclouds.profitbricks.rest.domain.Volume.Request.creatingBuilder()
+                    .dataCenterId(dataCenterId)
+                    .image(image.getId())
```

Comment:
Is it intentional that here we provision every volume with an image? (i.e.: If we attached 5 volumes, each has a bootable OS?)